core/src/main/scala/kafka/log/Log.scala

```diff
@@ -185,7 +185,7 @@ class Log(@volatile var dir: File,
   /* the actual segments of the log */
   private val segments: ConcurrentNavigableMap[java.lang.Long, LogSegment] = new ConcurrentSkipListMap[java.lang.Long, LogSegment]
 
-  val leaderEpochCache: LeaderEpochCache = initializeLeaderEpochCache()
+  var leaderEpochCache: LeaderEpochCache = initializeLeaderEpochCache()
```

Comment:
Does this not need to be `volatile`? Also, now that it's a `var`, we may want to make it private and only provide a public accessor.

Reply:
Good point. You are right, it should be volatile. I have updated the patch to make it private and added a public accessor for it.